tests/infra_test.go

```diff
@@ -975,3 +969,32 @@ func getKeysFromMetrics(metrics map[string]float64) []string {
 	sort.Strings(keys)
 	return keys
 }
+
+func checkError(err error) error {
+	var e *neturl.Error
+	if errors.As(err, &e) && e.Timeout() {
+		return nil
+	} else {
+		return err
+	}
+}
```

Comment:
I do not think you need this whole function, you just need the condition from it...
`errors.As(err, &e) && e.Timeout()`

So you can change it to `isTimeoutError`and simplify it or embed it in the caller one.